src/transport/SecurePairingSession.cpp

```diff
@@ -56,6 +58,87 @@ SecurePairingSession::~SecurePairingSession()
+    uint16_t mPeerKeyId;
+    uint64_t padding;
+} SecurePairingSessionSerializable;
+
+CHIP_ERROR SecurePairingSession::Serialize(SecurePairingSessionSerialized & output)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    const NodeId localNodeId = (mLocalNodeId.HasValue()) ? mLocalNodeId.Value() : kUndefinedNodeId;
+    const NodeId peerNodeId  = (mPeerNodeId.HasValue()) ? mPeerNodeId.Value() : kUndefinedNodeId;
+    VerifyOrExit(CanCastTo<uint16_t>(mKeLen), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(CanCastTo<uint64_t>(localNodeId), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(CanCastTo<uint64_t>(peerNodeId), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(CanCastTo<uint16_t>(sizeof(SecurePairingSessionSerializable)), error = CHIP_ERROR_INTERNAL);
+
+    {
+        uint8_t paired                                = (mPairingComplete) ? 1 : 0;
+        uint16_t keLen                                = static_cast<uint16_t>(mKeLen);
+        uint16_t serializedLen                        = 0;
+        SecurePairingSessionSerializable serializable = {
+            keLen, { 0 }, paired, localNodeId, peerNodeId, mLocalKeyId, mPeerKeyId, 0
+        };
+        memcpy(serializable.mKe, mKe, kMAX_Hash_Length);
+
+        VerifyOrExit(BASE64_ENCODED_LEN(sizeof(serializable)) <= sizeof(output.inner), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+        serializedLen = chip::Base64Encode(Uint8::to_const_uchar((uint8_t *) &serializable),
+                                           static_cast<uint16_t>(sizeof(serializable)), Uint8::to_char(output.inner));
+        VerifyOrExit(serializedLen > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrExit(serializedLen < sizeof(output.inner), error = CHIP_ERROR_INVALID_ARGUMENT);
+        output.inner[serializedLen] = '\0';
+    }
+
+exit:
+    return error;
+}
+
+CHIP_ERROR SecurePairingSession::Deserialize(SecurePairingSessionSerialized & input)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    SecurePairingSessionSerializable serializable;
+    size_t maxlen            = BASE64_ENCODED_LEN(sizeof(serializable));
+    size_t len               = strnlen(Uint8::to_char(input.inner), maxlen);
+    uint16_t deserializedLen = 0;
+
+    VerifyOrExit(BASE64_MAX_DECODED_LEN(len) <= sizeof(serializable), error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(CanCastTo<uint16_t>(len), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    memset(&serializable, 0, sizeof(serializable));
+    deserializedLen =
+        Base64Decode(Uint8::to_const_char(input.inner), static_cast<uint16_t>(len), Uint8::to_uchar((uint8_t *) &serializable));
+    VerifyOrExit(deserializedLen > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(deserializedLen <= sizeof(serializable), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    mPairingComplete = (serializable.mPairingComplete == 1);
+
+    VerifyOrExit(CanCastTo<uint16_t>(serializable.mKeLen), error = CHIP_ERROR_INTERNAL);
+    mKeLen = static_cast<size_t>(serializable.mKeLen);
```

Comment:
Why check for castability to `uint16_t` if we then cast to `size_t`?

Reply:
Copy-pasta mistake (from serialize). `serializable.mKeLen` is actually `uint16_t`, so check is redundant. I'll remove it. Thanks for catching it.